### ISSUE TITLE
Merging of feat/mana branch

### ISSUE THREAD

**Issue by lzpap** [iotaledger/goshimmer#810]
Mana currently lives in `feat/mana` branch that should be merged back into develop once merging of the message/value tangle is complete.
During the merge, pay attention to:
- [x] Add new transaction fields (pledge IDs + tx timestamp) to new transaction models.
- [x] Add time range check on tx timestamp.
- [x] Faucet changes (pledging mana + output pre-mining).
- [x] Client lib: new api `allowedManaPledge`.
- [x] Wallet lib: ask node about pledgeIDs before constructing a new transaction.
- [x] Integration test framework changes.
- [x] Local dashboard mana changes, generate new packr.
- [x] Rearrange grafana dashboard with included mana panels.
- [x] Add default mana config values to config file.
- [x] Mana plugin using the new (merged) tangle events.
- [x] Add more integration tests
- [ ] Update comments on any mana methods that take the optional `updateTime` parameter.
       **APIs**
    - [x] mana/all
    - [x] /mana/access/nhighest (consensus too)
    - [x] /mana/percentile
    - [x] /mana/access/online (consensus too)
    - [x] /mana/pending
    - [x] /mana/consensus/past
    - [x] /mana/consensus/logs
    - [x] /mana/consensus/metadata

nice to haves:
 - [ ] nodeID and tx ID in mana event pop-up is clickable
 - [x] API documentation (wiki)
 - [x] Communication between wallet and node to actually negotiate pledge ID.

**Comment by acha-bill** [comment]
This is going to be rough
